File: api/src/main/java/com/project/supershop/features/orders/enums/OrderStatus.java
package com.project.supershop.features.orders.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Arrays;

@Getter
@AllArgsConstructor
public enum OrderStatus {
    PENDING("pending"),
    CONFIRMED("confirmed"),
    DELIVERING("delivering"),
    COMPLETED("completed"),
    CANCELLED("cancelled"),
    REFUNDED("refunded");
    private final String value;

    public String value() {
        return this.value;
    }

    public static boolean isValidOrderStatus(String value) {
        for (OrderStatus status : OrderStatus.values()) {
            if (status.value().equals(value)) {
                return true;
            }
        }
        return false;
    }
}

File: app/lib/enums/order_status.dart
enum OrderStatus {
  pending,
  confirmed,
  delivering,
  completed,
  cancelled,
  refunded,
  all,
}

OrderStatus getOrderStatus(String status) {
  switch (status) {
    case 'pending':
      return OrderStatus.pending;
    case 'confirmed':
      return OrderStatus.confirmed;
    case 'delivering':
      return OrderStatus.delivering;
    case 'completed':
      return OrderStatus.completed;
    case 'cancelled':
      return OrderStatus.cancelled;
    case 'refunded':
      return OrderStatus.refunded;
    case 'all':
      return OrderStatus.all;
    default:
      throw Exception('Unknown order status: $status');
  }
}
